lib/smart_proxy_monitoring_icinga2/monitoring_icinga2_main.rb

```diff
@@ -78,7 +78,7 @@ def set_downtime_host(host, author, comment, start_time, end_time, all_services:
     private
 
     def uri_encode_filter(filter)
-      URI.encode(filter)
+      CGI.escape(filter)
```

Comment:
CGI.escape is quite a different API from URI.encode. I opened https://github.com/theforeman/smart_proxy_monitoring/pull/30 and would prefer to investigate that path instead.

Reply:
I changed the `required_ruby_version` from` '>= 2.5', '< 3'` to `'>= 2.7', '< 4'` and we are not getting any URI.escape error any more. So I move back to the previous change.

Reply:
You can see that the URL that's stubbed is changed, and it introduces a problem.

In the old version the parameters are separated by `&` but in the new version that's now escaped and represented as `%26%26`. This means only two parameters (`filter` and `type`) are sent, instead of all 4 (`filter`, `author`, `comment`, `type`).